### PR TITLE
CASMINST-4643

### PR DESCRIPTION
**Pull request by cdelatte-hpe**
# Description

Adding ceph container vesion upgrade check and guidance as to how to interpret the check

# Checklist Before Merging

- [ ] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [ ] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [ ] My commits or Pull-Request Title contain my JIRA information, or I don't have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/MTL-1695/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: upgrade/1.2/Stage_4.md [hunk 1]
```diff
@@ -22,6 +22,38 @@ of the first three storage nodes (`ncn-s001`, `ncn-s002`, or `ncn-s003`).
...
+
+      "target_id": "cba763a65a95e8849d578e05b111123f55a78ab096e67e8ecf7fdc98e67aea71",
+      "target_name": "registry.local/artifactory.algol60.net/csm-docker/stable/quay.io/ceph/  ceph:v15.2.15",
+      "target_version": "ceph version 15.2.15 (2dfb18841cfecc2f7eb7eb2afd65986ca4d95985) octopus   (stable)",
+   ```
+
+   **NOTE:**
+   * This upgrade is targeting the Ceph processes running 15.2.8 only.
+   * The monitoring services may be listed but those are patched internally and will not be upgraded with this upgrade.  This will include `alertmanager, prometheus, node-exporter, and grafana`.
+   * The main goal of this check is to see the listed 15.2.8 services and to see the output at the bottom that is confirming the presence of the image.
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_4.md:55]:
```suggestion
   * The main goals of this check are to see the listed 15.2.8 services and to see the output at the bottom that confirms the presence of the 15.2.15 target image.
```

File: upgrade/1.2/Stage_4.md [hunk 2]
```diff
@@ -22,6 +22,51 @@ of the first three storage nodes (`ncn-s001`, `ncn-s002`, or `ncn-s003`).
...
+           "crash.ncn-s001": {
+               "current_id": "6a777b4f888c24feec6e12eeeff4ab485f2c043b415bc2213815d5fb791f2597",
+               "current_name": "registry.local/ceph/ceph:v15.2.8",
+               "current_version": "15.2.8"
+           },
+           "crash.ncn-s002": {
+               "current_id": "6a777b4f888c24feec6e12eeeff4ab485f2c043b415bc2213815d5fb791f2597",
+               "current_name": "registry.local/ceph/ceph:v15.2.8",
+               "current_version": "15.2.8"
+           },
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_4.md:44]:
```suggestion
```

File: upgrade/1.2/Stage_4.md [hunk 3]
```diff
@@ -22,6 +22,49 @@ of the first three storage nodes (`ncn-s001`, `ncn-s002`, or `ncn-s003`).
    ceph orch upgrade check --image registry.local/artifactory.algol60.net/csm-docker/stable/quay.io/ceph/ceph:v15.2.15
    ```
 
+   Example output:
+
+   ```json
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_4.md:27]:
```suggestion
   ```text
```

File: upgrade/1.2/Stage_4.md [hunk 4]
```diff
@@ -22,6 +22,49 @@ of the first three storage nodes (`ncn-s001`, `ncn-s002`, or `ncn-s003`).
...
+               "current_version": "15.2.8"
+           },
+           "crash.ncn-s002": {
+               "current_id": "6a777b4f888c24feec6e12eeeff4ab485f2c043b415bc2213815d5fb791f2597",
+               "current_name": "registry.local/ceph/ceph:v15.2.8",
+               "current_version": "15.2.8"
+           },
+   ```
+
+   Omitting lines for readability. Expected output ends with lines similar to the following:
```
Comment (nrockershousen) [inline upgrade/1.2/Stage_4.md:46]: Not required, but usually we use `[...]` surrounded by blank lines in the code block to show omitted output. 

For example:

    ```json
    {
        "needs_update": {
            "alertmanager.ncn-s001": {
                "current_id": "6ec9fa439af31102c9e6581cbb3d12ee2ab258dada41d40d0f8ad987e8ff266f",
                "current_name": "registry.local/quay.io/prometheus/alertmanager:v0.21.0",
                "current_version": "0.21.0"
            },
            "crash.ncn-s001": {
                "current_id": "6a777b4f888c24feec6e12eeeff4ab485f2c043b415bc2213815d5fb791f2597",
                "current_name": "registry.local/ceph/ceph:v15.2.8",
                "current_version": "15.2.8"
            },
            "crash.ncn-s002": {
                "current_id": "6a777b4f888c24feec6e12eeeff4ab485f2c043b415bc2213815d5fb791f2597",
                "current_name": "registry.local/ceph/ceph:v15.2.8",
                "current_version": "15.2.8"
            },
    
      [...]
  
            "rgw.site1.zone1.ncn-s003.adrubu": {
                "current_id": "6a777b4f888c24feec6e12eeeff4ab485f2c043b415bc2213815d5fb791f2597",
                "current_name": "registry.local/ceph/ceph:v15.2.8",
                "current_version": "15.2.8"
            }
        },
        "target_id": "cba763a65a95e8849d578e05b111123f55a78ab096e67e8ecf7fdc98e67aea71",
        "target_name": "registry.local/artifactory.algol60.net/csm-docker/stable/quay.io/ceph/ceph:v15.2.15",
        "target_version": "ceph version 15.2.15 (2dfb18841cfecc2f7eb7eb2afd65986ca4d95985) octopus (stable)",
        "up_to_date": []
    }
    ```

File: upgrade/1.2/Stage_4.md [hunk 5]
```diff
@@ -22,6 +22,49 @@ of the first three storage nodes (`ncn-s001`, `ncn-s002`, or `ncn-s003`).
...
+
+  ```json
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_4.md:48]:
```suggestion
  ```text
```

File: upgrade/1.2/Stage_4.md (hunk 4 above)
Comment (mharding-hpe) [reply upgrade/1.2/Stage_4.md:47]: Yeah, I wasn't sure how the json markup would handle that, but it definitely doesn't like what I've been trying, so let me try that.

File: upgrade/1.2/Stage_4.md [hunk 6]
```diff
@@ -22,6 +22,49 @@ of the first three storage nodes (`ncn-s001`, `ncn-s002`, or `ncn-s003`).
...
+               "current_name": "registry.local/ceph/ceph:v15.2.8",
+               "current_version": "15.2.8"
+           },
+           "crash.ncn-s002": {
+               "current_id": "6a777b4f888c24feec6e12eeeff4ab485f2c043b415bc2213815d5fb791f2597",
+               "current_name": "registry.local/ceph/ceph:v15.2.8",
+               "current_version": "15.2.8"
+           },
+
+           [...],
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_4.md:46]:
```suggestion
           "[ ... lines omitted for readability ... ]",
```

**Review by nrockershousen — APPROVED** [review]

**Comment by mharding-hpe** [comment]
Closing this in favor of Brad's PR, which incorporated the changes from this one.
https://github.com/Cray-HPE/docs-csm/pull/1856

**Pull request closed without merging**